File: Discord_win/Discord_win/Models/Role.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Discord.Models {
    public class Role {
        public int RoleId { get; set; }
        public int RoleLevel { get; set; }
        public bool MainRole { get; set; }
        public string RoleName { get; set; }
        public bool Kick { get; set; }
        public bool ModifyChannel { get; set; }
        public bool ModifyRole { get; set; }
        public bool ChangeUserRole { get; set; }
        public int ServerId { get; set; }

        public Server Server { get; set; }
        public ICollection<ChannelPermission> ChannelPermissions { get; set; }
        public static bool operator <(Role a, Role b) {
            return a.RoleLevel < b.RoleLevel;
        }
        public static bool operator <=(Role a, Role b) {
            return a.RoleLevel <= b.RoleLevel;
        }
        public static bool operator >(Role a, Role b) {
            return a.RoleLevel > b.RoleLevel;
        }
        public static bool operator >=(Role a, Role b) {
            return a.RoleLevel >= b.RoleLevel;
        }
    }
}

File: Discord_android/Discord/app/src/main/java/com/peanut/discord/models/Role.java
package com.peanut.discord.models;
import java.util.List;
public class Role {
    private int roleId;
    private int roleLevel;
    private boolean mainRole;
    private String roleName;
    private boolean kick;
    private boolean manageChannel;
    private boolean manageRole;
    private boolean changeUserRole;
    private int serverId;
    private Server server;
    private List<ChannelPermission> channelPermissions;
    public int getRoleId() {
        return roleId;
    }
    public void setRoleId(int roleId) {
        this.roleId = roleId;
    }
    public int getRoleLevel() {
        return roleLevel;
    }
    public void setRoleLevel(int roleLevel) {
        this.roleLevel = roleLevel;
    }
    public boolean isMainRole() {
        return mainRole;
    }
    public void setMainRole(boolean mainRole) {
        this.mainRole = mainRole;
    }
    public String getRoleName() {
        return roleName;
    }
    public void setRoleName(String roleName) {
        this.roleName = roleName;
    }
    public boolean isKick() {
        return kick;
    }
    public void setKick(boolean kick) {
        this.kick = kick;
    }
    public boolean isManageChannel() {
        return manageChannel;
    }
    public void setManageChannel(boolean manageChannel) {
        this.manageChannel = manageChannel;
    }
    public boolean isManageRole() {
        return manageRole;
    }
    public void setManageRole(boolean manageRole) {
        this.manageRole = manageRole;
    }
    public boolean isChangeUserRole() {
        return changeUserRole;
    }
    public void setChangeUserRole(boolean changeUserRole) {
        this.changeUserRole = changeUserRole;
    }
    public int getServerId() {
        return serverId;
    }
    public void setServerId(int serverId) {
        this.serverId = serverId;
    }
    public Server getServer() {
        return server;
    }
    public void setServer(Server server) {
        this.server = server;
    }
    public List<ChannelPermission> getChannelPermissions() {
        return channelPermissions;
    }
    public void setChannelPermissions(List<ChannelPermission> channelPermissions) {
        this.channelPermissions = channelPermissions;
    }
}
